src/NServiceBus.Core/Unicast/MessageOperations.cs

```diff
@@ -109,6 +116,8 @@ Task SendMessage(IBehaviorContext context, Type messageType, object message, Sen
                 options.Context,
                 context);
 
+            outgoingContext.Set("MessageOperationContext", new ContextBag(options.MessageOperationContext));
```

Comment:
also needs to be applied here: https://github.com/Particular/NServiceBus/pull/6316/files#diff-6da20f5dc0d3f50670c1454e728e6ad6463f1c270868a5d2eb307784fb16835cR157